reviews/review-01+2020-11-02.md

```diff
@@ -0,0 +1,305 @@
+"syllabus" of the book. Its aim would be strengthened by adding a
+section on "How to Use this Book" or expanding the Syllabus (1.3) and/or
+Audience (1.2) sections with:
+
+-   Suggestions on how the book could be used. This could be as simple
+    as -- "Work through each chapter, following along with the commands
+    and changes described in the text. Try at least half of the
+    exercises for each chapter" -- to something as complex as an
+    appendix with suggestions for a suggested course plan, adapting
+    these materials to a lecture, finding a mentor or someone to help
+    you (if doing self-study), etc.
+
+-   Emphasizing the presence of exercises and solutions, as well as
+    recommendations for how to use them. I agree with the decision to
+    include the exercises at the end of each chapter, but I think a note
+    recommending that readers not wait until the end to check out some
+    of the available exercises would be a good idea.
+
+-   Outlining the standard chapter format: introduction, content,
+    exercises; identifying the callout box format as a mechanism to
+    supplement the main flow of text.
+
+My only additional comments are about rearranging/splitting specific
+material or chapters. None of these are strictly necessary -- I leave it
+up to the discretion of the authors as to whether they would improve the
+overall manuscript or not.
+
+-   The project organization description (1.4) seemed a bit out of place
+    in chapter 1 -- most of that chapter is about course **details**
+    (audience, installation, syllabus), but the Project Structure
+    section reads like course **content**. If the above suggestion is
+    added, this will even more be the case. I see a compelling argument
+    for making the current chapter 1 into a numberless prologue or
+    introduction and having chapter 1 be a very short chapter with the
+    Project Structure content.
+
+-   Chapter 2 (The Basics of the Unix Shell) is very long. Content-wise,
+    there isn't a strong reason to split (although the separation
+    between 2.8 and 2.9 is one possible place to do so); I would mainly
+    recommend splitting this chapter so that there isn't so much content
+    before getting to exercises.
+
+-   The section on package documentation (13.6) seems like it could be
+    its own chapter, as it delves fairly deeply into the nuts and bolts
+    of package documentation tools, which is a departure from the rest
+    of the chapter's focus on packaging as a whole. I can understand
+    wanting to include the documentation section in the packaging
+    chapter so it's not an afterthought, but it makes for a chapter that
+    covers two quite distinct topics (conceptually) -- packaging, and
+    good documentation practices. Another option besides splitting the
+    packaging chapter (ch 13) into two would be to put some of the more
+    detailed information about documentation generation into an
+    appendix.
```

Comment:
I agree that we need a re-think of Section 13.6 (Package Documentation). Breaking it down into its component parts:
- **13.6**: The opening section introduces the differences between novices, competent practitioners and experts as well as other documentation concepts like a chorus of explanations. None of this information is used in the simple zipf package documentation page that we create throughout the chapter, so I think we could probably move this content to an appendix and refer readers to it (i.e. we create a basic ReadTheDocs page during the chapter and then tell people to read the appendix to find out how to add more complete / comprehensive documentation to that page).
-  **13.6.1 Writing good docstrings**: This probably belongs back in Section 4.3, where we actually first meet and use docstrings.
- **13.6.2 to 13.6.4**: These are the sections where we use sphinx and ReadTheDocs to create the documentation webpage. I think they can probably stay in the packaging chapter as they are.
- **13.6.5 FAQs**: We don't actually add any FAQs to our simple documentation webpage, so this section could be put with the opening section in an appendix.

Assuming we create this new appendix with more detailed information about writing documentation, one of the exercises in this chapter could be to read that appendix and add more detailed information to the skeleton sphinx docs we create during the chapter.

Reply:
Agree here. Regarding the last paragraph about the exercise to read the appendix, I think that if that's the case, we could make it as a section in the book that they read as part of the exercise rather than as an appendix. I suggest this because to me, an appendix is meant as something to use "if interested" or "supplemental", but if we make it part of the exercise, it no longer is "an appendix", it is a part of the course. But, given the exercises are at the end rather than interspersed, this would be difficult to implement at this stage... Anyway, just my thoughts.

Reply:
I think this is a good plan too. I like the idea of moving the docstring material into one place, and since it is a short section it would not disrupt the flow of that chapter.

Reply:
@lwjohnst86 I agree that appendix information should be supplemental and not required. In this case, I guess I'd be inclined to define content that is only needed to complete an exercise as supplemental.